boot/zephyr/main.c

```diff
@@ -202,6 +201,8 @@ static void do_boot(struct boot_rsp *rsp)
 
 #if CONFIG_CPU_HAS_ARM_MPU || CONFIG_CPU_HAS_NXP_MPU
     z_arm_clear_arm_mpu_config();
+#else
+    irq_lock();
```

Comment:
So the irq_lock() is not invoked when the CPU has MPU. Why? What is the relation between cleaning the IRQ state and whether the SoC has the MPU peripheral? I think these two are unrelated.

Reply:
Miss aligned the #else will fix it.